README.md

```diff
@@ -222,6 +222,62 @@ It will output the following formatted code:
 Set the `RUST_LOG=debug` environment variable if you want to enable
 debug logging.
 
+## Configuration
+Topiary is configured using `languages.toml` files. There are three
+locations where Topiary checks for such a file.
+
+### Locations
+At buildtime the [languages.toml](./languages.toml) in the root of
+this repository is included into Topiary. This file is parsed at
+runtime. The purpose of this `languages.toml` file is to provide sane
+defaults for users of Topiary (both the library and the binary).
+
+The other two are read by the Topiary binary at runtime and allow the user to
+configure Topiary to their needs. The first is intended to be user specific, and
+can thus be found in the configuration directory of the OS:
+```
+Unix: /home/alice/.config/topiary/languages.toml
+Windows: C:\Users\Alice\AppData\Roaming\Topiary\config\languages.toml
+MacOS: /Users/Alice/Library/Application Support/Topiary/languages.toml
+```
+This file is not automatically created by Topiary.
+
+The last location is intended to be a project-specific settings file for
+Topiary. When running Topiary in some directory, it will look up in the file
+tree until it finds a .topiary directory. It will then read the `languages.toml`
+file present in that directory.
+
+The Topiary binary parses these file in the following order, any configuration
+options defined earlier are overwritten by those defined later.
+
+1. The builtin configuration file
+2. The user configuration file in the OS's configuration directory
+3. The project specific topiary configuration
+
+### Configuration Options
+The configuration file is contains a list of languages, each language
+configuration headed by the ``[[language]]`` toml header. For instance, the one
```

Comment:
I don't particularly like "headed by the header" 🙂

Reply:
That's fair  :) Lemme fix that